dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNServiceManagerInterfaceIIDM.cpp

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright (c) 2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+/**
+ * @file  DYNServiceManagerInterfaceIIDM.cpp
+ *
+ * @brief Service manager : implementation file for IIDM implementation
+ *
+ */
+
+#include "DYNServiceManagerInterfaceIIDM.h"
+
+#include "DYNDataInterfaceIIDM.h"
+#include "DYNNetworkInterfaceIIDM.h"
+#include "DYNSwitchInterfaceIIDM.h"
+
+namespace DYN {
+
+ServiceManagerInterfaceIIDM::ServiceManagerInterfaceIIDM(const DataInterfaceIIDM* const dataInterface) : dataInterface_(dataInterface), graphs_{} {}
+
+void
+ServiceManagerInterfaceIIDM::buildGraph(const boost::shared_ptr<VoltageLevelInterface>& vl) {
+  auto compare = [](const boost::shared_ptr<BusInterface>& bus, const std::string& id) { return bus->getID() == id; };
+
+  auto& graph = graphs_[vl->getID()];
+  const auto& buses = vl->getBuses();
+
+  for (size_t i = 0; i < buses.size(); i++) {
+    // we are using the position in the vector as the vertex, as these buses are fixed
+    graph.addVertex(i);
```

Comment:
couldn't we create a dictionary here to avoid  using find_if below?

Reply:
Using this way allows to reuse DYN graph component, which is much easier and avoids code duplicata

Reply:
yes, I mean a dictionary bus-id=> index to avoid searching several time the index of each bus into the vector

Reply:
OK